drivers/clock_control/clock_control_rpi_pico.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2022 Andrei-Edward Popa
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT raspberrypi_pico_clocks
+
+#include <zephyr/drivers/clock_control/rpi_pico_clock_control.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/reset.h>
+
+/* pico-sdk includes */
+#include <hardware/clocks.h>
+
+#define CTRL_ENABLE_BITS CLOCKS_CLK_GPOUT0_CTRL_ENABLE_BITS
+#define FIXED_CLOCK_FREQ(node_id, child) DT_PROP(DT_CHILD(node_id, child), clock_frequency)
```

Comment:
This seems to be written specifically to access the children of `clocks`, so maybe it should be rewritten to something like:
```
#define FIXED_CLOCK_FREQ(clock) DT_PROP(DT_CHILD(DT_NODELABEL(clocks), clock), clock_frequency)
```